File: src/CourseService/Application/Courses/Dtos/ManagedCourseSummaryDto.cs
using Courses.Application.Categories.Dtos;
using Courses.Application.Services.LinkProvider.Abstractions;
using Courses.Application.Shared.Dtos;
using Courses.Domain.Courses.Primitives;
using Kernel;

namespace Courses.Application.Courses.Dtos;

public record ManagedCourseSummaryDto
{
    public required Guid Id { get; init; }
    public required string Title { get; init; }
    public required string ShortDescription { get; init; }
    public required string Slug { get; init; }

    public required InstructorDto Instructor { get; init; }
    public required CategoryDto Category { get; init; }

    public required Money Price { get; init; }
    public required DifficultyLevel Difficulty { get; init; }

    public required string? ThumbnailUrl { get; init; }
    public required DateTimeOffset UpdatedAtUtc { get; init; }

    public required CourseStatus Status { get; init; }

    public required ManagedCourseStatsDto Stats { get; init; }

    public required IReadOnlyList<LinkDto> Links { get; init; }
}

public sealed record ManagedCourseStatsDto(
    int LessonsCount,
    TimeSpan Duration
);

File: Frontend/src/domain/courses/types/ManagedCourseSummaryDto.ts
import type { LinkDto } from "@/shared/types/LinkDto";
import type { Money } from "./money";
import type { CategoryDto, CourseStatus, InstructorDto } from "./CourseSummaryDto";
import type { ManagedCourseSummaryLinks } from "./links";

/** Data-only shape returned inside item.data by GET /manage/courses */
export interface ManagedCourseSummaryDataDto {
  id: string;
  title: string;
  shortDescription: string;
  slug: string;
  instructor: InstructorDto;
  category: CategoryDto;
  price: Money;
  difficulty: string;
  thumbnailUrl: string | null;
  updatedAtUtc: string;
  status: CourseStatus;
  stats: ManagedCourseStatsDto;
}

/** Backend returns items as { data, links }; view model flattens to this for components */
export interface ManagedCourseSummaryDto {
  id: string;
  title: string;
  shortDescription: string;
  slug: string;
  instructor: InstructorDto;
  category: CategoryDto;
  price: Money;
  difficulty: string;
  thumbnailUrl: string | null;
  updatedAtUtc: string;
  status: CourseStatus;
  stats: ManagedCourseStatsDto;
  /** Legacy array or strongly-typed record from API */
  links: LinkDto[] | ManagedCourseSummaryLinks;
}

/** Raw item shape from GET /manage/courses (backend sends data + links) */
export interface ManagedCourseSummaryItemDtoApi {
  data: ManagedCourseSummaryDataDto;
  links: ManagedCourseSummaryLinks;
}


export interface ManagedCourseStatsDto {
  lessonsCount: number;
  duration: string;
}
